nilearn/glm/tests/test_first_level.py

```diff
@@ -1958,6 +1958,25 @@ def test_first_level_from_bids_no_subject(tmp_path):
         )
 
 
+def test_first_level_from_bids_unused_kwargs(tmp_path):
+    """Check that unused kwargs are properly handled."""
+    bids_path = create_fake_bids_dataset(
+        base_dir=tmp_path, n_sub=1, n_ses=1, tasks=["main"], n_runs=[2]
+    )
+    with pytest.raises(
+        RuntimeError, match="Some keyword arguments were unused:"
```

Comment:
Just realised this would have to be updated as well
```suggestion
        RuntimeError, match="Unknown keyword arguments"
```